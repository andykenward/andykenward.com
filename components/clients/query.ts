import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const FRAGMENT_CLIENT: DocumentNode = gql`
  fragment FragmentClient on Client {
    id
    title
    href
  }
`
