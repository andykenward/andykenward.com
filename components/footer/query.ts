import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const QUERY_FOOTER: DocumentNode = gql`
  query Footer {
    footer(where: { id: "cjzjk2d1gvch50887pxsk6yju" }) {
      id
      links {
        ...Link
      }
    }
  }
`
