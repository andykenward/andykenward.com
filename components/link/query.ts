import gql from "graphql-tag"

gql`
  fragment Link on Link {
    id
    title
    url
  }
`
