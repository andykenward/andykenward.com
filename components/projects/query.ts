import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const QUERY_PROJECTS: DocumentNode = gql`
  query Projects {
    projects {
      id
      slug
      title
    }
  }
`

gql`
  fragment ProjectNavItem on Project {
    id
    clients {
      ...Clients
    }
    slug
    title
    description
    role
    studio {
      id
      title
      url
    }
    tags {
      ...Tags
    }
    urlCaseStudy
    url
  }
`

export const QUERY_PROJECTS_NAV: DocumentNode = gql`
  query ProjectsNav {
    projects {
      ...ProjectNavItem
    }
  }
`
