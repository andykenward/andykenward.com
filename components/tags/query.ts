import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const FRAGMENT_TAG: DocumentNode = gql`
  fragment FragmentTag on Tag {
    id
    title
    href
  }
`
