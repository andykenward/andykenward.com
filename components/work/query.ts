import { DocumentNode } from "graphql"
import gql from "graphql-tag"

export const QUERY_WORK: DocumentNode = gql`
  query Work {
    projects(orderBy: date_DESC) {
      __typename
      id
      slug
      title
    }
  }
`
