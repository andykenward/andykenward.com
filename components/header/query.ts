import { DocumentNode, FragmentSpreadNode } from "graphql"
import gql from "graphql-tag"

import { FRAGMENT_LINK } from "../link/query"

export const FRAGMENT_ME: FragmentSpreadNode = gql`
  fragment FragmentMe on Me {
    id
    name
    role
    update
    links {
      ...FragmentLink
    }
  }
  ${FRAGMENT_LINK}
`

export const QUERY_ME: DocumentNode = gql`
  query Me {
    me(where: { id: "cjzkx8reiebu10941932j2b3x" }) {
      ...FragmentMe
    }
  }
  ${FRAGMENT_ME}
`
