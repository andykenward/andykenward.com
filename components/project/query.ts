import { DocumentNode } from "graphql"
import gql from "graphql-tag"

import { FRAGMENT_PROJECTNAV } from "../projects/query"

export const QUERY_PROJECTS: DocumentNode = gql`
  query Project($slug: String) {
    project(where: { slug: $slug }) {
      ...FragmentProjectNav
    }
  }
  ${FRAGMENT_PROJECTNAV}
`
