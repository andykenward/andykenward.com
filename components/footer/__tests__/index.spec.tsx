import { useQuery } from "@apollo/react-hooks"
import { MockedResponse, wait } from "@apollo/react-testing"
import { RenderResult } from "@testing-library/react"
import { act } from "react-dom/test-utils"

import { Footer } from ".."
import { renderWithApp } from "../../../lib/test-utils"
import { QUERY_FOOTER } from "../query"
import { FOOTER_RESULT_DATA } from "./__fixtures__"

describe("<Footer />", () => {
  const MOCKS: MockedResponse[] = [
    {
      request: {
        query: QUERY_FOOTER,
      },
      result: {
        data: FOOTER_RESULT_DATA,
      },
    },
  ]

  it("should handle Footer query", async () => {
    const Component = () => {
      const { data, loading } = useQuery(QUERY_FOOTER)
      if (!loading) {
        expect(data).toEqual(FOOTER_RESULT_DATA)
      }
      return null
    }
    await act(async () => {
      renderWithApp(<Component />, MOCKS)
    })
  })

  it("renders null on loading", async () => {
    let root: RenderResult

    act(() => {
      root = renderWithApp(<Footer />, MOCKS)
    })

    act(() => {
      const { container } = root
      expect(container.firstChild).toBeNull()
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it("renders with 5 items", async () => {
    let root: RenderResult
    await act(async () => {
      root = renderWithApp(<Footer />, MOCKS)
    })

    await act(async () => {
      await wait(0)
    })

    act(() => {
      const { getAllByTestId } = root
      expect(getAllByTestId("footer-item").length).toEqual(5)
      expect(getAllByTestId("footer-item-a").length).toEqual(5)
      expect(root.container.firstChild).toMatchSnapshot()
    })
  })
})
