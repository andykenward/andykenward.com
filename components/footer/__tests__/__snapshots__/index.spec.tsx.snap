// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders null on loading 1`] = `null`;

exports[`<Footer /> renders with 5 items 1`] = `
.emotion-22 {
  padding-top: 2rem;
  padding-bottom: 2rem;
  display: grid;
  grid-template-columns: 32px [center-start] 1fr [center-end] 32px;
}

@media screen and (min-width:40em) {
  .emotion-22 {
    grid-template-columns: minmax(32px,15%) [center-start] 1fr [center-end] minmax(32px,15%);
  }
}

.emotion-20 {
  margin: 0;
  padding: 0;
  display: grid;
  grid-row-gap: 1rem;
  grid-column: center-start/center-end;
  list-style: none;
}

.emotion-2 {
  margin: 0;
  padding: 0;
  display: list-item;
}

.emotion-0 {
  font-family: Graphik,"Helvetica Neue",Helvetica,Arial,Verdana,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: rgb(36,28,21);
  margin: 0;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:visited:hover,
.emotion-0:hover,
.emotion-0:focus {
  color: #007c89;
}

.emotion-0:visited {
  color: #808080;
}

<footer
  class="emotion-22 emotion-3"
>
  <ul
    class="emotion-20 emotion-3"
  >
    <li
      class="emotion-2 emotion-3"
      data-testid="footer-item"
    >
      <a
        class="emotion-0 emotion-1"
        data-testid="footer-item-a"
        href="https://github.com/andykenward"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      data-testid="footer-item"
    >
      <a
        class="emotion-0 emotion-1"
        data-testid="footer-item-a"
        href="mailto:hello@andykenward.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Email
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      data-testid="footer-item"
    >
      <a
        class="emotion-0 emotion-1"
        data-testid="footer-item-a"
        href="https://www.linkedin.com/in/andykenward"
        rel="noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      data-testid="footer-item"
    >
      <a
        class="emotion-0 emotion-1"
        data-testid="footer-item-a"
        href="https://stackoverflow.com/story/andykenward"
        rel="noopener noreferrer"
        target="_blank"
      >
        Stack Overflow
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      data-testid="footer-item"
    >
      <a
        class="emotion-0 emotion-1"
        data-testid="footer-item-a"
        href="https://codepen.io/andykenward/"
        rel="noopener noreferrer"
        target="_blank"
      >
        CodePen
      </a>
    </li>
  </ul>
</footer>
`;
